relay/limiter/chunk_rate_limiter.go

```diff
@@ -36,12 +37,17 @@ type ChunkRateLimiter struct {
 	// perClientOperationsInFlight is the number of GetChunk operations currently in flight for each client.
 	perClientOperationsInFlight map[string]int
 
+	// limiterCounter is used to track rate limiting events, ignored if nil
+	limiterCounter metrics.CountMetric
+
 	// this lock is used to provide thread safety
 	lock sync.Mutex
 }
 
 // NewChunkRateLimiter creates a new ChunkRateLimiter.
-func NewChunkRateLimiter(config *Config) *ChunkRateLimiter {
+func NewChunkRateLimiter(
+	config *Config,
+	limiterCounter metrics.CountMetric) *ChunkRateLimiter {
```

Comment:
nit: it's kind of strange that it's only taking in the count metric. The metrics a component is tracking probably shouldn't be defined by constructor arguments

Reply:
I've refactored this. Instead of taking an individual metric, it now accepts an instance of the `RelayMetrics` wrapper that contains all relay metrics.